src/utils.jl

```diff
@@ -20,3 +20,18 @@ function wcs_to_celestial_frame(wcs::WCSTransform)
 
     return radesys
 end
+
+"""
+    pad_edges(array_in::Matrix{T}) where {T}
+
+Pads a given array and creates a border with edge elements.
+"""
+function pad_edges(array_in::Matrix{T}) where {T}
+    image = zeros(size(array_in)[1] + 2, size(array_in)[2] + 2)
```

Comment:
You're not using the type `T`.  I'm pretty sure that for type-stability you really want to allocate a matrix with eltype `T`